gradle.properties

```diff
@@ -24,5 +24,13 @@ systemProp.defaultSparkVersions=3.5
 systemProp.knownSparkVersions=3.3,3.4,3.5
 systemProp.defaultScalaVersion=2.12
 systemProp.knownScalaVersions=2.12,2.13
+# enable the Gradle build cache
+org.gradle.caching=true
+# enable Gradle parallel builds
 org.gradle.parallel=true
+# configure only necessary Gradle tasks
+org.gradle.configureondemand=true
+# explicitly disable the configuration cache
+org.gradle.unsafe.configuration-cache=false
```

Comment:
The meaning of the configuration is something a reader will be able to look up. These comments are OK, but it would be great to capture the reasoning -- the rationale why we're setting these configuration the way we're setting them. This is important in case someone thinks it would be good to change them further. The rationale informs what to watch out for.

Reply:
+1 on adding some reasoning as to why certain things are enabled